nixos/tests/cntr.nix

```diff
@@ -0,0 +1,38 @@
+# Test for cntr tool
+
+import ./make-test-python.nix ({pkgs, ...}: {
+  name = "cntr";
+
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ sorki mic92 ];
+  };
+
+  machine = { lib, ... }: {
+    environment.systemPackages = [ pkgs.cntr ];
+    containers.test = {
+      autoStart = true;
+      privateNetwork = true;
+      hostAddress = "172.16.0.1";
+      localAddress = "172.16.0.2";
+      config = {};
+    };
+
+    virtualisation.oci-containers = {
+      backend = "docker";
+      containers.nginx = {
+        image = "nginx-container";
+        imageFile = pkgs.dockerTools.examples.nginx;
+        ports = ["8181:80"];
+      };
+    };
+  };
+
+  testScript = ''
+    machine.start()
+    machine.wait_for_unit("container@test.service")
+    machine.wait_for_unit("docker-nginx.service")
+    machine.succeed("cntr attach test sh -- -c 'ping -c5 172.16.0.1'")
+    machine.succeed("cntr attach -t docker nginx sh -- -c 'ip a'")
```

Comment:
I think it'd make sense to actually write an assertion against the output.

Reply:
That should do:

```suggestion
    machine.succeed("cntr attach -t docker nginx sh -- -c 'ip a' | grep -q 172.16.0.2") 
```

Reply:
Added assert for output and switched to `curl localhost` as podman/docker use different subnets.